### ISSUE TITLE
Parameters with same name in camel- and snake-case should not be permitted

### ISSUE THREAD

**Issue by gheine** [apicollective/apibuilder#684]
Apibuilder currently accepts when an operation has the same parameter multiple times, if they are e.g. in snake and camel case (e.g. someId and some_id). Unfortunately it looks like all code generators generate invalid code (e.g. functions with the same parameter name twice). 

Example: 

```
{
  "name": "test",
  "base_url": "http://localhost.com",
  "apidoc": {
    "version": "0.11.48"
  },
  "models": {
    "test": {
      "fields": [
        {
          "name": "id",
          "type": "string"
        }
      ]
    }
  },
  "resources": {
    "test": {
      "operations": [
        {
          "method": "GET",
          "path": "/foo/:someId/bar/:some_id",
          "parameters": [
            {
              "name": "someId",
              "type": "string",
              "location": "path"
            },
            {
              "name": "some_id",
              "type": "string",
              "location": "path"
            }
          ],
          "responses": {
            "204": { "type": "unit" }
          }
        },
        {
          "method": "GET",
          "path": "/foo/:someId",
          "parameters": [
            {
              "name": "someId",
              "type": "string",
              "location": "path"
            },
            {
              "name": "some_id",
              "type": "string",
              "location": "query"
            }
          ],
          "responses": {
            "204": { "type": "unit" }
          }
        }
      ]
    }
  }
}
```

**Comment by gheine** [comment]
Example: 

```
{
  "name": "test",
  "base_url": "http://localhost.com",
  "apidoc": {
    "version": "0.11.48"
  },
  "models": {
    "test": {
      "fields": [
        {
          "name": "id",
          "type": "string"
        }
      ]
    }
  },
  "resources": {
    "test": {
      "operations": [
        {
          "method": "GET",
          "path": "/foo/:someId/bar/:some_id",
          "parameters": [
            {
              "name": "someId",
              "type": "string",
              "location": "path"
            },
            {
              "name": "some_id",
              "type": "string",
              "location": "path"
            }
          ],
          "responses": {
            "204": { "type": "unit" }
          }
        },
        {
          "method": "GET",
          "path": "/foo/:someId",
          "parameters": [
            {
              "name": "someId",
              "type": "string",
              "location": "path"
            },
            {
              "name": "some_id",
              "type": "string",
              "location": "query"
            }
          ],
          "responses": {
            "204": { "type": "unit" }
          }
        }
      ]
    }
  }
}
```

**Comment by gheine** [comment]
Fixed in #690
